azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java

```diff
@@ -24,8 +24,9 @@
  */
 public interface PartitionReceiver {
 
-    int MINIMUM_PREFETCH_COUNT = 10;
-    int DEFAULT_PREFETCH_COUNT = 999;
+    int MINIMUM_PREFETCH_COUNT = 1;
+    int DEFAULT_PREFETCH_COUNT = 500;
+    int MAXIMUM_PREFETCH_COUNT = 2000;
```

Comment:
>2000 [](start = 33, length = 4)

we used to have a max and then later took it off; motivation was that - we already had service checks and spark adapter wanted to put higher values here.
I am unsure about exact values but, i know that, spark adapter sets this close to ~2k; can you please check this with Sabee.

Reply:
I've already made a change in the Spark adapter and set the max to 2000. :-)